src/lib/templating.ts

```diff
@@ -263,7 +271,11 @@ export class PipelineInterpolator implements StepMatcher<S.PipelineStep> {
   }
 
   join(step: Readonly<S.JoinStep>) {
-    return { ...step };
+    if (S.isNotDeReferenced(step.right_pipeline)) {
+      return { ...step };
+    } else {
+      return { ...step, right_pipeline: this.interpolate(step.right_pipeline) };
+    }
```

Comment:
Is there a case when we arrive here and the pipeline is still referenced?
Maybe for validation when we submit the step form?

I suggest we indicate when this happens, so further readers will understand why the two cases are necessary.
(and if not... remove one :D)

Reply:
Bonne idée! https://github.com/ToucanToco/weaverbird/pull/654/commits/7891b0e4be90b70dafe86daf14f6817b06c05065